File: EstruturaDados/src/InsertSort.java
public class InsertSort {
    public static void insert(int[] array) {
        for (int i = 1; i < array.length; i++) {
            int chave = array[i];
            int j = i - 1;
            while (j >= 0 && array[j] > chave) {
                array[j + 1] = array[j];
                j--;
            }
            array[j + 1] = chave;
        }
    }
    public static void main(String[] args) {
        int[] array = {5, 2, 4, 6, 1, 3};
        insert(array);
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
    }
}

File: EstruturaDados/src/insertSort.py
def insertionSort(lista):
    for i in range(1, len(lista)):
        chave = lista[i]
        j = i - 1
        while j >= 0 and chave < lista[j]:
            lista[j + 1] = lista[j]
            j -= 1
        lista[j + 1] = chave
    return lista

lista = [12, 11, 13, 5, 6]
print(insertionSort(lista))